app/routes/task_routes.py

```diff
@@ -0,0 +1,124 @@
+import requests
+import os
+from datetime import datetime
+from flask import Blueprint, jsonify, abort, make_response, request
+from dotenv import load_dotenv
+from app import db
+from app.models.task import Task
+from .routes_helpers import validate_model, find_date
+
+load_dotenv()
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+
+# CREATE ENDPOINT
+@task_bp.route("", methods=["POST"])
+def create_task():
+    req_body = request.get_json()
+
+    if "title" in req_body and "description" in req_body:
+        if "completed_at" in req_body:
+            date = find_date(req_body["completed_at" ])
+            new_task = Task.from_dict(req_body, date)
+        else:
+            new_task = Task.from_dict(req_body)
+    else:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response(jsonify({"task": new_task.to_dict()}), 201)
+
+
+# GET ALL ENDPOINT
+@task_bp.route("", methods=["GET"])
+def handle_tasks():
+    sort_param = request.args.get("sort")
+    
+    if sort_param == "asc":
+        task_query = Task.query.order_by(Task.title.asc())
+    elif sort_param == "desc":
+        task_query = Task.query.order_by(Task.title.desc())
+    else:
+        task_query = Task.query.all()
+
+    tasks_response = [task.to_dict() for task in task_query]
+
+    return jsonify(tasks_response), 200
+
+
+# GET ONE ENDPOINT
+@task_bp.route("/<id>", methods=["GET"])
+def handle_task(id):
+    task = validate_model(Task, id)
+
+    return jsonify({"task": task.to_dict()}), 200
+
+
+# UPDATE ONE ENDPOINT
+@task_bp.route("/<id>", methods=["PUT"])
+def update_task(id):
+    task = validate_model(Task, id)
+
+    req_body = request.get_json()
+
+    task.title = req_body["title"]
+    task.description = req_body["description"]
+
+    if "completed_at" in req_body:
+        date = find_date(req_body["completed_at" ])
+        task.completed_at = date
+
+    db.session.commit()
+
+    return make_response(jsonify({"task": task.to_dict()}), 200)
+
+
+# DELETE ONE ENDPOINT
+@task_bp.route("/<id>", methods=["DELETE"])
+def delete_task(id):
+    task = validate_model(Task, id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details": f"Task {task.task_id} \"{task.title}\" successfully deleted"}, 200)
+
+
+# MARK TASK AS COMPLETE
+@task_bp.route("/<id>/mark_complete", methods=["PATCH"])
+def mark_completed(id):
+    task = validate_model(Task, id)
+    prev_incomplete = True if not task.completed_at else False
```

Comment:
This seems more like a toggle between marked complete or incomplete. That would work if the route's purpose was for toggling. But this route is specific to marking a task complete and only that. We have a route that's only for marking a task incomplete.

We shouldn't overlap, so consider removing line 94 completely, as well as line 99. Even if this task was already marked complete, that's fine! This specific route's sole functionality is to change the `completed_at` timestamp and call the slack API!